src/main/java/org/ow2/proactive/procci/service/CloudAutomationInstanceClient.java

```diff
@@ -56,95 +47,64 @@
+     *  Give an optional containing a instance model if the parameters match with an instance in cloud automation
+     * @param variableName is a key in the variables for the cloud automation model
+     * @param variableValue is a value in the variables for the cloud automation model
+     * @param transformerProvider is a transformer for converting the cloud automation model into a instance model
+     * @return an instance model if the parameters match otherwise return an empty optional
      */
-    public JSONObject postRequest(JSONObject content) {
-
-        final String PCA_SERVICE_SESSIONID = "sessionid";
-        final String url = requestUtils.getProperty(INSTANCE_ENDPOINT);
-        try {
-            CloseableHttpClient httpClient = HttpClientBuilder.create().build();
-            HttpPost postRequest = new HttpPost(url);
-            postRequest.addHeader(PCA_SERVICE_SESSIONID, requestUtils.getSessionId());
-            StringEntity input = new StringEntity(content.toJSONString());
-            input.setContentType("application/json");
-            postRequest.setEntity(input);
-
-            HttpResponse response = httpClient.execute(postRequest);
-
-            String serverOutput = requestUtils.readHttpResponse(response, url, "POST " + content.toJSONString());
-            httpClient.close();
-            return parseJSON(serverOutput);
-        } catch (IOException ex) {
-            logger.error(" IO exception in CloudAutomationInstanceClient::postRequest " + ", exception : " +
-                         ex.getMessage());
-            throw new ServerException();
-        }
+    public Optional<InstanceModel> getInstanceModel(String variableName, String variableValue,
+            TransformerProvider transformerProvider) {
+        return getInstanceByVariable(variableName,
+                                     variableValue).map(model -> transformerProvider.toInstanceModel(model));
     }
 
-    private JSONObject parseJSON(String jsonString) {
-        try {
-            return (JSONObject) new JSONParser().parse(jsonString);
-        } catch (ParseException ex) {
-            logger.error(" Parse exception in CloudAutomationInstanceClient::parseJSON " + ", exception : " +
-                         ex.getMessage());
-            throw new ServerException();
-        }
-
+    /**
+     * Create an instance in cloud automation from instanceModel
+     * @param instanceModel is the model that is used to create the instance
+     * @param actionType is the action to apply on the instance
+     * @param transformerProvider is the transformer to apply on the instance model
+     * @return a instance model return by cloud automation
+     */
+    public InstanceModel postInstanceModel(InstanceModel instanceModel, String actionType,
+            TransformerProvider transformerProvider) {
+        return transformerProvider.toInstanceModel(new Model(requestUtils.postRequest(transformerProvider.toCloudAutomationModel(instanceModel,
```

Comment:
Because the transformation is the business  value bring by multi-language connector